homeassistant/components/flo/device_action.py

```diff
@@ -0,0 +1,106 @@
+"""Provides device actions for Flo by Moen devices."""
+from typing import List
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_DEVICE_ID, CONF_DOMAIN, CONF_TYPE
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+
+from .const import DOMAIN as FLO_DOMAIN
+from .device import FloDeviceDataUpdateCoordinator
+from .services import (
+    ATTR_DEVICE_ID,
+    ATTR_LOCATION_ID,
+    ATTR_REVERT_TO_MODE,
+    ATTR_SLEEP_MINUTES,
+    SERVICE_RUN_HEALTH_TEST,
+    SERVICE_SET_AWAY_MODE,
+    SERVICE_SET_HOME_MODE,
+    SERVICE_SET_SLEEP_MODE,
+    SYSTEM_MODE_HOME,
+)
+
+FLO_ACTION_TYPE_SERVICE_CALL = "service_call"
+
+ACTION_SCHEMA = cv.DEVICE_ACTION_BASE_SCHEMA.extend(
+    {vol.Required(CONF_DOMAIN): FLO_DOMAIN, vol.Required(CONF_TYPE): str}
+)
+
+DEVICE_ACTIONS = [
+    {CONF_TYPE: SERVICE_RUN_HEALTH_TEST, CONF_DOMAIN: FLO_DOMAIN},
+    {CONF_TYPE: SERVICE_SET_AWAY_MODE, CONF_DOMAIN: FLO_DOMAIN},
+    {CONF_TYPE: SERVICE_SET_HOME_MODE, CONF_DOMAIN: FLO_DOMAIN},
+    {CONF_TYPE: SERVICE_SET_SLEEP_MODE, CONF_DOMAIN: FLO_DOMAIN},
```

Comment:
Maybe make this a `SET_MODE` and add a dropdown for away/home/sleep via the capabilities ?

Reply:
Oh I see now that sleep mode is a standalone thing. Away/home are related.